UnityProject/Assets/Scripts/ModManager.cs

```diff
@@ -176,14 +204,50 @@ public class ModManager : MonoBehaviour {
 
         // Generate Mod Object
         var mod = new Mod(assetPath);
-        mod.name = bundleName;
+        mod.name = Path.GetFileName(bundleName);
         mod.modType = GetModTypeFromBundle(modBundle);
 
+        // Determine gun display name for the cache
+        if(mod.modType == ModType.Gun)
+            mod.name = modBundle.LoadAsset<GameObject>(ModManager.GetMainAssetName(ModType.Gun)).GetComponent<WeaponHolder>().display_name;
+
         // Register mod and clean up
         availableMods.Add(mod);
         modBundle.Unload(true);
         Debug.Log($" + {bundleName} ({mod.modType})");
     }
+
+    private static void LoadCache() {
+        string path = Path.Combine(ModManager.GetModsfolderPath(), "cache");
+
+        if(File.Exists(path))
+            availableMods = new List<Mod> (JsonUtility.FromJson<Cache>(File.ReadAllText(path)).mods);
+        else
+            availableMods = new List<Mod> ();
+    }
+
+    private static void UpdateCache() {
+        try {
+            string path = Path.Combine(ModManager.GetModsfolderPath(), "cache");
+
+            if(File.Exists(path))
+                File.Delete(path);
+
+            File.Create(path).Close();
+            File.WriteAllText(path, JsonUtility.ToJson(new Cache(availableMods.ToArray())));
+        } catch (Exception e) {
+            Debug.LogError(e);
+        }
+    }
```

Comment:
I am a little worried this part doesn't work properly on Linux or Mac systems. But I guess people will say something if it ends up not working?

Reply:
Looks ok to me on a first looks, case is an important thing to keep in mind which seems covered.